src/gpu/matrix/KmMatrix/Generator.cuh

```diff
@@ -0,0 +1,100 @@
+/*!
+ * Copyright 2018 H2O.ai, Inc.
+ * License   Apache License Version 2.0 (see LICENSE for details)
+ */
+
+#include <thrust/random.h>
+#include <random>
+
+#include <curand_kernel.h>
+
+#include "Generator.hpp"
+#include "KmMatrix.hpp"
+#include "../../utils/utils.cuh"
+
+
+namespace h2o4gpu {
+namespace Matrix {
+
+namespace kernel {
+// Split the definition to avoid multiple definition.
+__global__ void setup_random_states(int _seed, curandState *_state,
+                                    size_t _size);
+
+__global__ void generate_uniform_kernel(float *_res,
+                                        curandState *_state,
+                                        int _size);
+
+__global__ void generate_uniform_kernel(double *_res,
+                                        curandState *_state,
+                                        int _size);
+}
+
+template <typename T>
+struct UniformGenerator : public GeneratorBase<T> {
```

Comment:
A few things:
- I think this should be named UniformRandomGenerator. Generator by itself does not imply that it generates random numbers.
- We should definitely remove the curand dependency if possible. Extra dependencies heavily impact maintainability.
- I think src/gpu/matrix/KmMatrix/Generator.hpp is unnecessary now that we have established we don't need to compile without cuda. Perhaps just put its contents in this file?

Reply:
> I think this should be named UniformRandomGenerator. Generator by itself does not imply that it generates random numbers.

Not problem. it will change.

> We should definitely remove the curand dependency if possible. Extra dependencies heavily impact maintainability.

Make sense, it will change.

> I think src/gpu/matrix/KmMatrix/Generator.hpp is unnecessary now that we have established we don't need to compile without cuda. Perhaps just put its contents in this file?

If we need to run it in CPU, the header is still useful if we create a CPU generator and put this file in src/common. That's something for future, is it acceptable for you?